src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala

```diff
@@ -217,14 +217,24 @@ class HttpGoogleBillingDAO(appName: String,
     ConfigContents(Seq(Resources(projectName, dmTemplatePath, properties))).toYaml.prettyPrint
   }
 
+  protected def whenDeploymentDeleteConflict(throwable: Throwable): Boolean = {
+    throwable match {
+      case t: GoogleJsonResponseException => t.getStatusCode == 409
+      case _ => false
+    }
+  }
+
   /*
  * Set the deployment policy to "abandon" -- i.e. allows the created project to persist even if the deployment is deleted --
  * and then delete the deployment. There's a limit of 1000 deployments so this is important to do.
  */
-  override def cleanupDeployment(projectName: String): Unit = {
+  override def cleanupDeployment(projectName: String): Future[Unit] = {
     if( cleanupDeploymentAfterCreating ) {
-      executeGoogleRequest(
+      retryExponentially(whenDeploymentDeleteConflict) { () => Future(blocking(executeGoogleRequest(
```

Comment:
Why the use of `blocking()` here?

Reply:
it gives the executor a hint on how to operate, see [here](https://docs.scala-lang.org/overviews/core/futures.html#blocking-inside-a-future). I copied it from the implementation of `retryWhen500OrGoogleError`.